server/app/assets/javascripts/session.ts

```diff
@@ -0,0 +1,395 @@
+            ?.textContent || 'Session successfully extended'
+        ToastController.showToastMessage({
+          id: 'session-extended-toast',
+          content: successText,
+          type: 'success',
+          duration: 3000,
+          canDismiss: true,
+          canIgnore: false,
+          condOnStorageKey: null,
+        })
+        this.checkAndSetTimer()
+      } else {
+        const errorText =
+          document.getElementById('session-extended-error-text')?.textContent ||
+          'Failed to extend session'
+        ToastController.showToastMessage({
+          id: 'session-extend-error-toast',
+          content: errorText,
+          type: 'error',
+          duration: 3000,
+          canDismiss: true,
+          canIgnore: false,
+          condOnStorageKey: null,
+        })
+      }
+    })
+
+    /**
+     * Handle modal button actions using event delegation:
+     * - Primary actions:
+     *   - For inactivity warning: Extends session by submitting form to /extend-session
+     *   - For total length warning: Initiates logout
+     * - Secondary/close actions:
+     *   - Hides the modal
+     *   - Resets warning visibility flags
+     *   - Rechecks timeout conditions
+     *
+     * Uses data attributes to identify buttons and their actions:
+     * - data-modal-type: Identifies which modal the button belongs to
+     * - data-modal-primary: Primary action button (extend/logout)
+     * - data-modal-secondary: Secondary action button (dismiss)
+     * - data-close-modal: Close button (X)
+     */
+    document.addEventListener('click', (event) => {
+      const target = event.target as HTMLElement
+      if (!target) return
+
+      const modalType = target.getAttribute('data-modal-type')
+      if (!modalType) return
+
+      if (target.hasAttribute('data-modal-primary')) {
+        if (modalType === 'session-inactivity-warning') {
+          // submit the extend session form to /extend-session endpoint.
+          const form = document.getElementById(
+            'extend-session-form',
+          ) as HTMLFormElement
+          form?.requestSubmit()
+        } else if (modalType === 'session-length-warning') {
+          this.logout()
+        }
+      } else if (
+        target.hasAttribute('data-modal-secondary') ||
+        target.hasAttribute('data-close-modal')
+      ) {
+        this.hideModal(modalType)
+        if (modalType === 'session-inactivity-warning') {
+          this.inactivityWarningShown = false
+        } else if (modalType === 'session-length-warning') {
+          this.totalLengthWarningShown = false
+        }
+        this.checkAndSetTimer()
+      }
+    })
+  }
+
+  /**
+   * Hides the specified modal by adding the 'is-hidden' class.
+   * Updates visibility tracking flags when hiding a modal.
+   *
+   * @param modalType Type of modal to hide ('session-inactivity-warning' or 'session-length-warning')
+   */
+  private static hideModal(modalType: string) {
```

Comment:
suggestion (non-blocking): move this method to `modal.ts` and call it `hideUswdsModal` or something like that so it can easily be used in other places.